core/math/face3.cpp

```diff
@@ -104,6 +104,7 @@ int Face3::split_by_plane(const Plane &p_plane, Face3 p_res[3], bool p_is_point_
 
 		if (below_count == 4) {
 
+			ERR_FAIL_COND_V(polygons_created >= 3, polygons_created);
```

Comment:
Can this really happen? We already have `ERR_FAIL_COND_V(above_count >= 4 && below_count >= 4, 0); //bug in the algo` at line 83, so it should not be possible to reach this `if` branch with `polygons_created == 3`.